src/rust/ide/src/controller/searcher.rs

```diff
@@ -618,6 +625,27 @@ impl Searcher {
                 Mode::NewNode {position} => self.add_example(&example,position).map(Some),
                 _                        => Err(CannotExecuteWhenEditingNode.into())
             }
+            Action::CreateNewProject => {
+                match self.ide.manage_projects() {
+                    Ok(_) => {
+                        let ide    = self.ide.clone_ref();
+                        let logger = self.logger.clone_ref();
+                        executor::global::spawn(async move {
+                            // We checked that manage_projects returns Some just a moment ago, so
+                            // unwrapping is safe.
```

Comment:
1. the "Some" is not correctly formatted
2. What does it mean "just a moment ago"?
3. The safety is not clearly seen in the code, imo this should be refactored.

Reply:
1. It's a comment, not documentation, so I don't see any need for formatting. But it should be `Ok` after the refactoring
2. You're in match checking if it returned Ok.
3. It's quite hard to refactor; already discussed with @mwu-tow : it would make the manage_project method return Rc, not a reference, and thus we couldn't return self what is quite handy here.